httpClient/src/main/java/org/jfrog/artifactory/client/httpClient/http/auth/PreemptiveAuthInterceptor.java

```diff
@@ -1,39 +1,72 @@
 package org.jfrog.artifactory.client.httpClient.http.auth;
 
-import org.apache.http.HttpException;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.client.HttpResponseException;
 import org.apache.http.HttpHost;
 import org.apache.http.HttpRequest;
 import org.apache.http.HttpRequestInterceptor;
+import org.apache.http.HttpStatus;
 import org.apache.http.auth.AuthScope;
 import org.apache.http.auth.AuthState;
 import org.apache.http.auth.Credentials;
 import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.HttpRequestWrapper;
 import org.apache.http.client.protocol.HttpClientContext;
 import org.apache.http.impl.auth.BasicScheme;
 import org.apache.http.protocol.HttpContext;
 import java.io.IOException;
-
+import java.net.URI;
+import java.net.URISyntaxException;
 /**
  * Request interceptor to perform preemptive authentication with http client.
  *
  * @author Yossi Shaul
  */
 public class PreemptiveAuthInterceptor implements HttpRequestInterceptor {
 
-    @Override
-    public void process(HttpRequest request, HttpContext context) throws HttpException, IOException {
-        HttpClientContext clientContext = HttpClientContext.adapt(context);
-        AuthState authState = clientContext.getTargetAuthState();
+	private static final String ORIGINAL_HOST_CONTEXT_PARAM = "original.host.context.param";
+
+	@Override
+	public void process(HttpRequest request, HttpContext context) throws IOException, HttpResponseException {
+		if (!shouldSetAuthScheme(request, context)) {
+			return;
+		}
+		HttpClientContext clientContext = HttpClientContext.adapt(context);
+		AuthState authState = clientContext.getTargetAuthState();
+		// If there's no auth scheme available yet, try to initialize it preemptively
+		if (authState.getAuthScheme() == null) {
+			String accessToken = clientContext.getUserToken(String.class);
+			if (StringUtils.isNotEmpty(accessToken)) {
+				request.addHeader("Authorization", "Bearer " + accessToken);
+			} else {
+				CredentialsProvider credsProvider = clientContext.getCredentialsProvider();
+				HttpHost targetHost = clientContext.getTargetHost();
+				Credentials creds = credsProvider
+						.getCredentials(new AuthScope(targetHost.getHostName(), targetHost.getPort()));
+				if (creds == null) {
+					throw new HttpResponseException(HttpStatus.SC_UNAUTHORIZED, "No credentials for preemptive authentication");
```

Comment:
You are no longer allowing unauthenticated requests here.
I suggest modifying the code as follows. Instead of:
```
				if (creds == null) {
					throw new HttpResponseException(HttpStatus.SC_UNAUTHORIZED, "No credentials for preemptive authentication");
				}
				BasicScheme authScheme = new BasicScheme();
				authState.update(authScheme, creds);
```
try doing:
```
				if (creds != null) {
					authState.update(new BasicScheme(), creds);
				}
``

Reply:
Correct. I was so wrong here overriding the original functionality of this file. Made the necessary changes. Thanks!